amd_openvx_extensions/amd_nn/src/slice_layer.cpp

```diff
@@ -1,345 +1,393 @@
-                    input_dims[0], input_dims[1], input_dims[2], input_dims[3],
-                    outputn_dims[0], outputn_dims[1], outputn_dims[2], outputn_dims[3]);
-        num_channels += outputn_dims[2];
-
-        //output tensor configuration
-        type = in_type;
-        ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[i], VX_TENSOR_DATA_TYPE, &type, sizeof(type)));
-        ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[i], VX_TENSOR_NUMBER_OF_DIMS, &num_dims, sizeof(num_dims)));
-        ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[i], VX_TENSOR_DIMS, outputn_dims, sizeof(outputn_dims)));
-
-        i++;
+
+    // create array of input/indices/output tensors
+    vx_tensor input_tensor[param_dims], indices_tensor[param_dims], output_tensor[param_dims];
+    
+    // create context & graph
+    vx_context context = vxCreateContext();
+    status = vxGetStatus((vx_reference)context);
+    if(status) {
+        std::cerr << "ERROR: vxCreateContext() failed (" << status << ")" << std::endl;
+        return -1;
     }
 
-    if(num_channels != input_dims[2]) return ERRMSG(VX_ERROR_INVALID_DIMENSION, "validate: slice: num_channels=%ld != input_dims[2]=%ld\n", num_channels, input_dims[2]);
+    vxLoadKernels(context, "vx_nn"); // load vx_nn kernels
 
-    return VX_SUCCESS;
-}
+    vx_graph graph = vxCreateGraph(context);
+    status = vxGetStatus((vx_reference)graph);
+    if(status) {
+        std::cerr << "ERROR: vxCreateGraph() failed (" << status << ")" << std::endl;
+        return -1;
+    }
 
-//! \brief The kernel target support callback.
-static vx_status VX_CALLBACK query_target_support(vx_graph graph, vx_node node,
-                                                  vx_bool use_opencl_1_2,              // [input]  false: OpenCL driver is 2.0+; true: OpenCL driver is 1.2
-                                                  vx_uint32& supported_target_affinity // [output] must be set to AGO_TARGET_AFFINITY_CPU or AGO_TARGET_AFFINITY_GPU or (AGO_TARGET_AFFINITY_CPU | AGO_TARGET_AFFINITY_GPU)
-                                                  )
-{
-    supported_target_affinity = AGO_TARGET_AFFINITY_GPU;
-    return VX_SUCCESS;
-}
+    // copy input tensor
+    status = vxMapTensorPatch((vx_tensor)parameters[0], 2, nullptr, nullptr, &map_id, stride, (void **)&fptr, VX_READ_ONLY, VX_MEMORY_TYPE_HOST, 0);
+    if(status) {
+        std::cerr << "ERROR: vxMapTensorPatch() failed for input tensor (" << status << ")" << std::endl;
+        return -1;
+    }
+    for(int i=0; i<8; i++) {
+        input.push_back((float)fptr[i]);
+    }
+    status = vxUnmapTensorPatch((vx_tensor)parameters[0], map_id);
+
+    // create tmp_input and copy input tensor
+    vx_tensor tmpi_tensor = vxCreateTensor(context, input_num_dims, input_dims, input_type, 0);
+    status = vxMapTensorPatch(tmpi_tensor, input_num_dims, nullptr, nullptr, &map_id, stride, (void **)&fptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST, 0);
```

Comment:
Codacy found an issue: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5932170)